lib/Sema/CSSimplify.cpp

```diff
@@ -4372,6 +4498,114 @@ ConstraintSystem::simplifyApplicableFnConstraint(
     return simplified;
   }
 
+  // Handle @dynamicCallable applications.
```

Comment:
This is a lot of new code for a single method -- maybe factor it out into a new method?